src/audio/igo_nr/igo_nr.c

```diff
@@ -43,46 +43,79 @@ DECLARE_SOF_RT_UUID("igo-nr", igo_nr_uuid,  0x696ae2bc, 0x2877, 0x11eb, 0xad, 0x
 
 DECLARE_TR_CTX(igo_nr_tr, SOF_UUID(igo_nr_uuid), LOG_LEVEL_INFO);
 
+static void igo_nr_lib_process(struct comp_data *cd)
+{
+	/* Pass through the active channel if
+	 * 1) It's not enabled, or
+	 * 2) hw parameter is not valid.
+	 */
+	if (!cd->process_enable[cd->config.active_channel_idx] || cd->invalid_param) {
+		memcpy_s(cd->out, IGO_FRAME_SIZE * sizeof(int16_t),
+			 cd->in, IGO_FRAME_SIZE * sizeof(int16_t));
+	} else {
+		IgoLibProcess(cd->p_handle,
+			      &cd->igo_stream_data_in,
+			      &cd->igo_stream_data_ref,
+			      &cd->igo_stream_data_out);
+	}
+}
+
 #if CONFIG_FORMAT_S16LE
 static void igo_nr_capture_s16(struct comp_data *cd,
 			       const struct audio_stream *source,
 			       struct audio_stream *sink,
-			       int32_t src_frames,
-			       int32_t sink_frames)
+			       int32_t frames)
 {
 	int32_t in_nch = source->channels;
 	int32_t out_nch = sink->channels;
+	int32_t min_nch = MIN(in_nch, out_nch);
```

Comment:
do we expect a channel mistmatch?

Reply:
I am not sure about it. But we could choose to pass through as more channels as possible.

Reply:
@lgirdwood thoughts, typically i have not seen this.